pcdsdevices/analog_signals.py

```diff
@@ -0,0 +1,167 @@
+import time
+import pcdsdevices.utils as key_press
+from ophyd import Device, Component as Cpt, EpicsSignal, EpicsSignalRO
+
+
+class Acromag(Device):
+    """
+    Class for Acromag analog input/ouput signals
+
+    Parameters:
+    -----------
+    prefix : str
+        The Epics base of the acromag
+
+    name : str
+        A name to prefer to the device
+    """
+    # Components for each channel
+    ao1_0 = Cpt(EpicsSignal, ":ao1:0", kind='normal')
+    ao1_1 = Cpt(EpicsSignal, ":ao1:1", kind='normal')
+    ao1_2 = Cpt(EpicsSignal, ":ao1:2", kind='normal')
+    ao1_3 = Cpt(EpicsSignal, ":ao1:3", kind='normal')
+    ao1_4 = Cpt(EpicsSignal, ":ao1:4", kind='normal')
+    ao1_5 = Cpt(EpicsSignal, ":ao1:5", kind='normal')
+    ao1_6 = Cpt(EpicsSignal, ":ao1:6", kind='normal')
+    ao1_7 = Cpt(EpicsSignal, ":ao1:7", kind='normal')
+    ao1_8 = Cpt(EpicsSignal, ":ao1:8", kind='normal')
+    ao1_9 = Cpt(EpicsSignal, ":ao1:9", kind='normal')
+    ao1_10 = Cpt(EpicsSignal, ":ao1:10", kind='normal')
+    ao1_11 = Cpt(EpicsSignal, ":ao1:11", kind='normal')
+    ao1_12 = Cpt(EpicsSignal, ":ao1:12", kind='normal')
+    ao1_13 = Cpt(EpicsSignal, ":ao1:13", kind='normal')
+    ao1_14 = Cpt(EpicsSignal, ":ao1:14", kind='normal')
+    ao1_15 = Cpt(EpicsSignal, ":ao1:15", kind='normal')
+
+    ai1_0 = Cpt(EpicsSignalRO, ":ai1:0", kind='normal')
+    ai1_1 = Cpt(EpicsSignalRO, ":ai1:1", kind='normal')
+    ai1_2 = Cpt(EpicsSignalRO, ":ai1:2", kind='normal')
+    ai1_3 = Cpt(EpicsSignalRO, ":ai1:3", kind='normal')
+    ai1_4 = Cpt(EpicsSignalRO, ":ai1:4", kind='normal')
+    ai1_5 = Cpt(EpicsSignalRO, ":ai1:5", kind='normal')
+    ai1_6 = Cpt(EpicsSignalRO, ":ai1:6", kind='normal')
+    ai1_7 = Cpt(EpicsSignalRO, ":ai1:7", kind='normal')
+    ai1_8 = Cpt(EpicsSignalRO, ":ai1:8", kind='normal')
+    ai1_9 = Cpt(EpicsSignalRO, ":ai1:9", kind='normal')
+    ai1_10 = Cpt(EpicsSignalRO, ":ai1:10", kind='normal')
+    ai1_11 = Cpt(EpicsSignalRO, ":ai1:11", kind='normal')
+    ai1_12 = Cpt(EpicsSignalRO, ":ai1:12", kind='normal')
+    ai1_13 = Cpt(EpicsSignalRO, ":ai1:13", kind='normal')
+    ai1_14 = Cpt(EpicsSignalRO, ":ai1:14", kind='normal')
+    ai1_15 = Cpt(EpicsSignalRO, ":ai1:15", kind='normal')
+
+    def initialize_mesh(self, sp_ch, rb_ch, scale=1000.0):
+        """
+        Setup the Acromag for mesh power supply usage
+
+        Parameters
+        ----------
+        sp_ch : int
+            Setpoint Acromag channel to which high voltage supply setpoint
+            is connected. Range is 0 to 15
+
+        rb_ch : int
+            Read back Acromag channel to which high voltage readback is
+            connected. Range is 0 to 15
+
+        scale : float, optional
+            Gain for high voltage supply to be controlled by the Acromag
+        """
+        self.scale = scale
+        self.rb_pv = getattr(self, 'ai1_%s' % rb_ch).pvname
```

Comment:
If you use the above methodology combining the `ao/ai` into a single `EpicsSignal`, you then only have to pick one of the signals for `mesh_raw` and you can do away with the `sp_pv` and `rb_pv`.